src/addrman.h

```diff
@@ -486,9 +489,15 @@ class CAddrMan
         mapAddr.clear();
     }
 
-    CAddrMan()
+    CAddrMan(bool deterministic = false,
+             int32_t consistency_check_ratio = DEFAULT_ADDRMAN_CONSISTENCY_CHECKS)
+        : m_consistency_check_ratio(consistency_check_ratio)
```

Comment:
I've moved the `insecure_rand` to the initializer list as suggested.

I'm not going to remove the `nKey` assignment in this PR. I'd like to remove the `Clear()` method from `CAddrMan` in a future PR. The only place it's used in the product code is if peers.dat deserialization fails, we'll call `Clear()`. I think it's better just to instantiate a new `CAddrMan` object in that case, since it's definitely possible to imagine new data being added to `CAddrMan` and not added to the `Clear()` method. It wasn't previously possible to instantiate a new `CAddrMan` since it was a global initialized before `main()`, but now that its lifetime is managed to `context.node`, we can just reset the unique ptr to a new instance.

Reply:
Maybe also adding `explicit` would make sense.

Reply:
Made the ctor explicit.